tests/test_image_vault.cpp

```diff
@@ -420,3 +448,19 @@ TEST_F(ImageVault, valid_remote_and_alias_returns_valid_image_info)
     EXPECT_THAT(image.original_release, Eq("16.04 LTS"));
     EXPECT_THAT(image.id, Eq("e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"));
 }
+
+TEST_F(ImageVault, http_download_returns_expected_image_info)
+{
+    HttpURLDownloader http_url_downloader;
+    mp::DefaultVMImageVault vault{hosts, &http_url_downloader, cache_dir.path(), data_dir.path(), mp::days{0}};
+
+    auto image_url{"http://www.foo.com/images/foo.img"};
+    mp::Query query{instance_name, image_url, false, "", mp::Query::Type::HttpDownload};
+
+    mp::VMImage image;
+    EXPECT_NO_THROW(image = vault.fetch_image(mp::FetchType::ImageOnly, query, stub_prepare, stub_monitor));
+
+    // Hash is based on image url
+    EXPECT_THAT(image.id, Eq("7404f51c9b4f40312fa048a0ad36e07b74b718a2d3a5a08e8cca906c69059ddf"));
+    EXPECT_THAT(image.release_date, Eq(default_last_modified.toString().toStdString()));
+}
```

Comment:
Wouldn't a second fetch attempt exercise the crash scenario more closely?

Reply:
I'm not trying to explicitly test the crash scenario with this test.  We didn't have a test at all for http download types, so this is just a general test for that download type.

Reply:
Thanks for adding the general test, but why not go one step further and actually test the scenario where we got the crash? Anyway, I am not blocking on this.